src/viam/sdk/common/proto_type.hpp

```diff
@@ -27,64 +27,82 @@ struct get_helper {
     }
 };
 
-template <>
-struct get_helper<AttributeMap> {
-    template <typename V>
-    static AttributeMap get(V& v) {
-        auto* const result = boost::get<AttributeMap>(&v);
-        return result ? *result : nullptr;
-    }
-
-    template <typename V>
-    static std::shared_ptr<const AttributeMap::element_type> get(const V& v) {
-        auto* const result = boost::get<AttributeMap>(&v);
-        return result ? *result : nullptr;
-    }
-};
-
-}  // namespace prototype_details
+}  // namespace value_type_details
 
-class ProtoType {
+class AttributeMap {
    public:
-    ProtoType() {
-        proto_type_ = boost::blank();
-    }
-
-    explicit ProtoType(bool b) : proto_type_(std::move(b)) {}
-    explicit ProtoType(std::string s) : proto_type_(std::move(s)) {}
-    explicit ProtoType(const char* c) : proto_type_(std::string(c)) {}
-    explicit ProtoType(int i) : proto_type_(std::move(i)) {}
-    explicit ProtoType(double d) : proto_type_(std::move(d)) {}
-    explicit ProtoType(AttributeMap m) : proto_type_(std::move(m)) {}
-    explicit ProtoType(std::vector<std::shared_ptr<ProtoType>> v) : proto_type_(std::move(v)) {}
-    explicit ProtoType(const google::protobuf::Value& value);
-
-    google::protobuf::Value proto_value();
-    friend bool operator==(const ProtoType& lhs, const ProtoType& rhs);
-
-    template <typename T>
-    auto get() {
-        return prototype_details::get_helper<T>::get(proto_type_);
-    }
-
-    template <typename T>
-    auto get() const {
-        return prototype_details::get_helper<T>::get(proto_type_);
-    }
+    class value_type;
+
+    using map_type = std::unordered_map<std::string, value_type>;
+
+    class value_type {
+       public:
+        template <typename T>
+        auto get() {
+            return value_type_details::get_helper<T>::get(value_);
```

Comment:
Without any specialization for `get_helper<T>`, these just decay into `boost::get`. But I think you may want new specializations? If a user asks for the `std::vector<value_type>` member, does it get you it, or do you get the wrapper? I'm not actually sure.